### ISSUE TITLE
Desktop-specific APIs

### ISSUE THREAD

**Issue by ptmt** [ptmt/react-native-desktop#4]
Implement only most common APIs. Other could be easily covered by plugins. 
- [ ] Application menu;
- [ ] Contexts menu;

**Comment by yanivtal** [comment]
Hey, great work with this project! What milestone are you expecting to add these in? Next release is 0.2.0's right? Would this be 0.3.0 stuff?

**Comment by ptmt** [comment]
I've started to implement a real product based on RND now, which will use most of these APIs internally. So eventually these will be merged into master. If you need some particular thing let me know.

**Comment by yanivtal** [comment]
That's great! Congrats. I'm very interested in the file writing API's. I'll keep an eye out for these hitting.

**Comment by Amnesthesia** [comment]
Any news on the file-writing API (or file-reading for that matter)? Been looking around trying to figure out how to read files and directories with no luck

**Comment by ptmt** [comment]
I believe it should be the plugin, like `react-native-fs`. See also https://github.com/skeletcode/skelet/blob/master/macos/skelet/Modules/FilesystemManager.m#L34 for an example, there are 3 methods, not perfect, though. What's you task?

**Comment by matpaul** [comment]
@ptmt what of these tasks has more priority ? 
Is resizing by dragging mean NSSplitView ?

**Comment by ptmt** [comment]
Yes, NSSplitView would be one of key component for desktop app.

The blocker-issue, though, is dealing with resizing due to a lot of views
in AppKit. https://github.com/ptmt/react-native-macos/pull/93 this solution
was a workaround, I don't sure if it needs to be merged. If I had enough
time, I would update RN for macOS with latest React Native (including Yarn,
Flow updates and etc.) and keep profiling this. If you interested in let me
know, I'll provide some details.

The other core issue is events system, like mouse events and such. See
problems like this https://github.com/ptmt/react-native-macos/issues/58

Also: https://github.com/ptmt/react-native-macos/issues/40 (that's
something that original React Native doesn't have as a concept).

**Comment by matpaul** [comment]
I'll start with implementation NSSplitView

I saw issue with resizing in init app (when resize - view background resize some delay)
I'm very interested in this project so yep, if you have some more details it will be good to provide it.

Concerning Event bug - when I test cursor changes, onMouseEnter and onMouseLeave work excellent - so need some information how to reproduce it. 
 
It will be very cool to merge with last react native, maybe after 0.40 released.

I think about implementation #40, but at this moment i think there are many other work ))

Also, how to provide documentation for custom components that not in React Native docs?

**Comment by stanlemon** [comment]
Are there any prototypes around the file open dialog that could be looked at for inspiration?

**Comment by ptmt** [comment]
@stanlemon https://github.com/skeletcode/skelet/blob/master/macos/skelet/Modules/DialogManager.m#L32

**Comment by nathggns** [comment]
Did you manage to look at menubar icons yet?

**Comment by doprdele** [comment]
Can you open files yet?

**Comment by ptmt** [comment]
Yes, see https://github.com/skeletcode/skelet/blob/master/macos/skelet/Modules/DialogManager.m#L32 for example. If you need to open it without a user consent there is a few examples like this https://github.com/ptmt/ratpack/commit/d1c2b2e90228f63629bb8714c4b941a9411a2438, but you’ll need to write ObjC
